src/main/java/htsjdk/variant/vcf/VCFUtils.java

```diff
@@ -157,21 +159,77 @@ public static VCFHeader withUpdatedContigs(final VCFHeader oldHeader, final File
                                                                   final File referenceFile) {
-        final List<VCFContigHeaderLine> lines = new ArrayList<VCFContigHeaderLine>();
+        final List<VCFContigHeaderLine> lines = new ArrayList<>();
         final String assembly = referenceFile != null ? getReferenceAssembly(referenceFile.getName()) : null;
         for ( final SAMSequenceRecord contig : refDict.getSequences() )
             lines.add(makeContigHeaderLine(contig, assembly));
         return lines;
     }
 
     private static VCFContigHeaderLine makeContigHeaderLine(final SAMSequenceRecord contig, final String assembly) {
-        final Map<String, String> map = new LinkedHashMap<String, String>(3);
+        final Map<String, String> map = new LinkedHashMap<>(3);
         map.put("ID", contig.getSequenceName());
         map.put("length", String.valueOf(contig.getSequenceLength()));
         if ( assembly != null ) map.put("assembly", assembly);
         return new VCFContigHeaderLine(map, contig.getSequenceIndex());
     }
 
+    /**
+     * This method creates a temporary VCF file and it's appropriately named index file, and will delete them on exit.
+     * @param prefix - The prefix string to be used in generating the file's name; must be at least three characters long
+     * @param suffix - The suffix string to be used in generating the file's name; may be null, in which case the suffix ".tmp" will be used
+     * @return A File object referencing the newly created temporary VCF file
+     * @throws IOException - if a file could not be created.
+     */
+    public static File createTemporaryIndexedVcfFile(final String prefix, final String suffix) throws IOException {
+        final File out = File.createTempFile(prefix, suffix);
+        out.deleteOnExit();
+        String indexFileExtension = null;
+        if (suffix.endsWith(IOUtil.COMPRESSED_VCF_FILE_EXTENSION)) {
+            indexFileExtension = IOUtil.COMPRESSED_VCF_INDEX_EXTENSION;
+        }
+        else if (suffix.endsWith(IOUtil.VCF_FILE_EXTENSION)) {
+            indexFileExtension = IOUtil.VCF_INDEX_EXTENSION;
+        }
+        if (indexFileExtension != null) {
+            final File indexOut = new File(out.getAbsolutePath() + indexFileExtension);
+            indexOut.deleteOnExit();
+        }
+        return out;
+    }
+
+    /**
+     * This method makes a copy of the input VCF and creates an index file for it in the same location.
+     * This is done so that we don't need to store the index file in the same repo
+     * The copy of the input is done so that it and its index are in the same directory which is typically required.
+     * @param vcfFile the vcf file to index
+     * @return File a vcf file (index file is created in same path).
+     */
+    public static File createTemporaryIndexedVcfFromInput(final File vcfFile, final String tempFilePrefix) throws IOException {
```

Comment:
Same as before...

Reply:
same here.

Reply:
The problem is VariantContextWriterBuilder.setOutputFile(File).... 

I don't want this PR to become a "get htsjdk to write in Path" as well....

Reply:
ok...